bin/import-dmn-diagram.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+set -eu
+workspace=${1}
+tenant_id=${2}
+product=${3}
+
+s2sSecret=${S2S_SECRET:-AABBCCDDEEFFGGHH}
+
+#if [[ "${env}" == 'prod' ]]; then
+#  s2sSecret=${S2S_SECRET_PROD}
+#fi
+
+serviceToken=$($(realpath $workspace)/bin/utils/idam-lease-service-token.sh civil_service \
+  $(docker run --rm toolbelt/oathtool --totp -b ${s2sSecret}))
+
+dmnFilepath="$(realpath $workspace)/resources"
+
+for file in $(find ${dmnFilepath} -name '*.dmn')
+do
+  uploadResponse=$(curl --insecure -v --silent -w "\n%{http_code}" --show-error -X POST \
+    ${CAMUNDA_BASE_URL:-http://localhost:9404}/engine-rest/deployment/create \
+    -H "Accept: application/json" \
+    -H "ServiceAuthorization: Bearer ${serviceToken}" \
+    -F "deployment-name=$(basename ${file})" \
+    -F "deploy-changed-only=true" \
+    -F "deployment-source=$product" \
+    ${tenant_id:+'-F' "tenant-id=$tenant_id"} \
+    -F "file=@${dmnFilepath}/$(basename ${file})")
+
+upload_http_code=$(echo "$uploadResponse" | tail -n1)
+upload_response_content=$(echo "$uploadResponse" | sed '$d')
+
+if [[ "${upload_http_code}" == '200' ]]; then
+  echo "$(basename ${file}) diagram uploaded successfully (${upload_response_content})"
+  continue;
+fi
+
+echo "$(basename ${file}) upload failed with http code ${upload_http_code} and response (${upload_response_content})"
+continue;
```

Comment:
Sorry I have picked it up directly from the wa-task config repo. But I guess this is being done so that we can get all the upload errors in one go rather than getting them one by one.

Reply:
If I'm reading it correctly, this script is being called in `Jenkinsfile_CNP` on line `63`. Al we'll be doing will be logging in Jenkins that the upload failed but we won't fail the deployment in any way.

If this is the intention we should store in a variable the failure and then just after the `done` call `exit 1` to exit in error.

As it stands, we'll just continue with the build pretending that everything is fine. I am not comfortable with it.

Reply:
Now resolved.